searchlib/src/vespa/searchlib/memoryindex/posting_iterator.cpp

```diff
@@ -58,80 +85,108 @@ PostingIterator<interleaved_features>::doSeek(uint32_t docId)
     }
 }
 
-template <bool interleaved_features>
+/**
+ * Search iterator over memory field index posting list.
+ *
+ * Template parameters:
+ *   - interleaved_features: specifies whether the wrapped posting list has interleaved features or not.
+ *   - unpack_normal_features: specifies whether to unpack normal features or not.
+ *   - unpack_interleaved_features: specifies whether to unpack interleaved features or not.
+ */
+template <bool interleaved_features, bool unpack_normal_features, bool unpack_interleaved_features>
+class PostingIterator : public PostingIteratorBase<interleaved_features> {
+public:
+    using ParentType = PostingIteratorBase<interleaved_features>;
+
+    using ParentType::ParentType;
+    using ParentType::_feature_decoder;
+    using ParentType::_feature_store;
+    using ParentType::_itr;
+    using ParentType::_matchData;
+    using ParentType::getDocId;
+    using ParentType::getUnpacked;
+    using ParentType::setUnpacked;
+
+    void doUnpack(uint32_t docId) override;
+};
+
+template <bool interleaved_features, bool unpack_normal_features, bool unpack_interleaved_features>
 void
-PostingIterator<interleaved_features>::doUnpack(uint32_t docId)
+PostingIterator<interleaved_features, unpack_normal_features, unpack_interleaved_features>::doUnpack(uint32_t docId)
 {
     if (!_matchData.valid() || getUnpacked()) {
         return;
     }
     assert(docId == getDocId());
     assert(_itr.valid());
     assert(docId == _itr.getKey());
-    datastore::EntryRef featureRef(_itr.getData().get_features());
-    _featureStore.setupForUnpackFeatures(featureRef, _featureDecoder);
-    _featureDecoder.unpackFeatures(_matchData, docId);
-    setUnpacked();
+    if (unpack_normal_features) {
+        datastore::EntryRef featureRef(_itr.getData().get_features());
+        _feature_store.setupForUnpackFeatures(featureRef, _feature_decoder);
+        _feature_decoder.unpackFeatures(_matchData, docId);
+        setUnpacked();
```

Comment:
The call to setUnpacked() should not depend on unpack_normal_features being true.

Reply:
Fixed, PTAL